// Copyright (C) 2020 The Android Open Source Project
//
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
// http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.
syntax = "proto3";

option java_multiple_files = true;
option java_package = "com.android.emulator.control";
option objc_class_prefix = "AEC";

package android.emulation.control;
import "google/protobuf/empty.proto";

// A simplistic echo service that can be used by unit tests to validate
// some of the grpc behavior.
service TestEcho {
  rpc streamEcho(Msg) returns (stream Msg) {}
  rpc echo(Msg) returns (Msg) {}
  rpc data(google.protobuf.Empty) returns (Msg) {}
  rpc streamData(google.protobuf.Empty) returns (stream Msg) {}
}

// A simple message that can be send back and forth.
message Msg {
  // A message that you wish to send.
  string msg = 1;

  // A counter that the server will increment.
  uint64 counter = 2;

  bytes data = 3;
}
